### DISCUSSION TITLE
Locally Build

### DISCUSSION THREAD

**Discussion by dincerkaan** [zitadel/zitadel#6289] [Q&A] [answered]
Hello, I want to install Zitadel completely locally and make some changes on the code and test it. However I don't know how to do this and the docs are pretty messy for me. Previously, in addition to many links, I also looked at (https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md) but couldn't make any progress..
I would be very happy if someone could help me with this.

**Comment by fforootd** [comment]
We just recently switched to a make file that should ease up the friction.

This section here outlines how to use it https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md#contribute

But be aware that we already track a problem for this here https://github.com/zitadel/zitadel/issues/6277

**Reply by dincerkaan** [reply]
Hi @fforootd thanks for the answer. 

I've already reviewed this place but it was quite complicated for me. I don't want to run the project through Docker, instead I just wanted to use the source code. Is this possible? If possible I will repeat each step

**Reply by fforootd** [reply]
You do not need to use docker for this the make file has commands like `make compile` https://github.com/zitadel/zitadel/blob/main/Makefile#L9 to locally build zitadel

**Reply by dincerkaan** [reply]
![image](https://github.com/zitadel/zitadel/assets/80959458/f6c716b1-119a-46be-a356-6ef148661907)

how can i get rid of this error

**Reply by fforootd** [reply]
You need to install buf https://buf.build/docs/installation/ which we use to generate the grcp stubs

**Reply by dincerkaan** [reply]
Hi @fforootd 
I downloaded the buf and it is active as you can see at the end of the terminal. But I keep getting core_api error. I've been stuck in the same place for 2 days with this error.. where am I missing?
![image](https://github.com/zitadel/zitadel/assets/80959458/d01b321a-0aa1-41f0-b0c2-bb281c2efb85)
![image](https://github.com/zitadel/zitadel/assets/80959458/3aac00e7-cef9-438b-af2e-527ecc5e7b79)

**Reply by fforootd** [reply]
Hm without sudo it looks like it calls buf but has problem creating the directory .artifacts in the current dir.

**Comment by dincerkaan** [comment, marked as answer]
hey @fforootd  and folks. I am sharing a txt file that I definitively solved the problem I was having. You can directly convert it to sh file and run it and the installation will be completed. It may take some time to complete.. But before running the commands
You have to type 'sudo nano ~/.bashrc' and add the following two PATHs at the bottom.

export PATH=/home/linuxbrew/.linuxbrew/bin
export PATH=$HOME/go/bin

when you're done, you can access it from localhost:8080.

There were countless bugs that I encountered, and that's how I was finally able to run the source code. Hope it helps you, have a nice day.

[curl httpsbinaries.cockroachdb.comc.txt](https://github.com/zitadel/zitadel/files/12283046/curl.httpsbinaries.cockroachdb.comc.txt)

**Reply by fforootd** [reply]
thank you for sharing this, I think it make sense to use this as a way of making sure that all needed deps/tools are installed.

**Reply by fforootd** [reply]
FYI we just released a devcontainer definition that contains everything one needs to get a working dev env for zitadel.

https://github.com/zitadel/zitadel/blob/main/.devcontainer/devcontainer.json
